doc/banner/banner_hld.md

```diff
@@ -0,0 +1,263 @@
+
+    revision 2023-05-18 {
+        description "First Revision";
+    }
+
+    container sonic-banner {
+
+        container BANNER_MESSAGE {
+
+            description "BANNER_MESSAGE part of config_db.json";
+
+            container MESSAGE {
+                leaf state {
+                    type string {
+                        pattern "enabled|disabled";
+                    }
+                    description "Banner feature state";
+                    default disabled;
+                }
+
+                leaf login {
+                    type string;
+                    description "Banner message displayed to user before login prompt";
+                    default "Debian GNU/Linux 11";
+                }
+
+                leaf motd {
+                    type string;
+                    description "Banner message displayed to user after login prompt";
+                    default "You are on
+                              ____   ___  _   _ _  ____
+                             / ___| / _ \| \ | (_)/ ___|
+                             \___ \| | | |  \| | | |
+                              ___) | |_| | |\  | | |___
+                             |____/ \___/|_| \_|_|\____|
+
+                             -- Software for Open Networking in the Cloud --
+
+                             Unauthorized access and/or use are prohibited.
+                             All access and/or use are subject to monitoring.
+
+                             Help:    https://sonic-net.github.io/SONiC/
+                             ";
+                }
+
+                leaf logout {
+                    type string;
+                    description "Banner message dispalyed to the users on logout";
+                    default "";
+                }
+            } /* end of container MESSAGE */
+        }
+        /* end of container BANNER_MESSAGE */
+    }
+    /* end of top level container */
+}
+/* end of module sonic-banner */
+```
+
+# 3 Test plan
+
+## 3.1 Unit tests
+
+1. Configure login banner message
+    1. Logout form system. Login again - expected to see configured message
+    before login prompt.
+    2. Do not save configuration and reboot device -  expected to see default message before login prompt.
+    3. Save configuration and reboot device -  expected to see configured message before login prompt.
+2. Configure message of the day
+    1. Logout form system. Login again - expected to see configured message
+    after login prompt.
+    2. Do not save configuration and reboot device -  expected to see default message after login prompt.
+    3. Save configuration and reboot device -  expected to see configured message after login prompt.
+3. Configure logout banner message
+    1. Logout form system - expected to see configured logout message.
+    2. Do not save configuration and reboot device. Logout from system after reboot - expected to see default message before login prompt.
```

Comment:
I hope that expected result shall be the default logout banner message after login + logout sequence

Reply:
yes